### DISCUSSION TITLE
user defined order in multiple levels

### DISCUSSION THREAD

**Discussion by ghost** [simolus3/drift#1487] [Q&A] [answered]
First, thanks for the great package you've built
i'm trying to build a todo app. like most production ready todo apps user will be able to create many tags/projects and each task can have many tags. Thus far it's simple, but my problem is reordering items(todos), user should be able to reorder tasks not only in main screen but for each individual tag/project.
[https://begriffs.com/posts/2018-03-20-user-defined-order.html](url)
above link explains different ways to keep track of item order. adding another column to task table and managing order for it isn't that complicated. But how can i do this for each tag? lets say i add index column to "tagTaskEntries"(which is used to manage many-to-many relation) as well to keep track of task order in tag level, how can i get maximum index of tasks with tag "x" for example? is there a way to check index on insert to see if it already exists and actually push that item(like inset in array)?
sorry if it's a little all over the place, i'm rather new to SQL

**Comment by simolus3** [comment, marked as answer]
> how can i get maximum index of tasks with tag "x" for example

In SQL, you would write `(SELECT MAX(ta."index") FROM tasks ta INNER JOIN tag_task_entries te ON te.task = ta.id  WHERE te.tag = x)` (well, you may have to change column and table names a bit).

There's a way to write that in Dart as well, I think something like this should work:

```dart
final maxForTag = subqueryExpression(selectOnly(tasks)
  ..addColumns([tagTasksEntries.index])
  ..join(innerJoin(tagTasksEntries, tagTasksEntries.task.equalsExp(tasks.id)))
  ..where(tagTasksEntries.task.equals(x))
);
```

To automatically use the latest id, you can insert a `TagTasksEntriesCompanion.custom` with the index set to the `maxForTag` expression above.
